src/content/tutorials/user-auth-with-auth0/_index.md

```diff
@@ -0,0 +1,772 @@
+
+![User info](/tutorials/user-auth-with-auth0/media/userInfo.png)
+
+### Logging out users
+
+While a user's authentication cookie expires after a day, you may want to offer the ability for a user to log out manually. This is quite simple to implement - instead of letting the cookie expire automatically, your Workers application should pass a `Set-cookie` header that nulls out the `cookieKey` we previously defined. Let's create a `logout` function in `workers-site/auth0.js`, and import it in `workers-site/index.js`, calling it when a user requests `/logout`:
+
+```js
+// workers-site/auth0.js
+
+export const logout = event => {
+  const cookieHeader = event.request.headers.get('Cookie')
+  if (cookieHeader && cookieHeader.includes(cookieKey)) {
+    return {
+      headers: {
+        'Set-cookie': `${cookieKey}=""; SameSize-Lax; Secure;`,
+      },
+    }
+  }
+  return {}
+}
+```
+
+```js
+// workers-site/index.js
+import { ..., logout } from './auth0.js'
+
+async function handleEvent(event) {
+  try {
+    // END OF WORKERS SITES CODE BLOCK
+
+    // BEGINNING OF LOGOUT CODE BLOCK
+    if (url.pathname === "/logout") {
+      const { headers } = logout(event);
+      return headers
+        ? new Response(response.body, {
+            ...response,
+            headers: Object.assign({}, response.headers, headers)
+          })
+        : Response.redirect(url.origin);
+    }
+    // END OF LOGOUT CODE BLOCK
+
+    // BEGINNING OF STATE HYDRATION CODE BLOCK
+  }
+}
+```
+
+In your Workers Site, you can add a "Log out" link, which will send users to the `/logout` route, and clear their auth cookie via the `Set-cookie` header:
+
+```html
+<a href="/logout">Log out</a>
+```
+
+Note that by design, the placement of the `LOGOUT CODE BLOCK` in `workers-site/index.js` is placed _after_ the `WORKERS SITES CODE BLOCK`. This is intentional: it allows users to continue to your application's deployed Workers Site code, meaning that you can provide a corresponding `logout/index.html` template with a "You're logged out!" message, or something similar. Here's an example you can use:
+
+```html
+<!-- public/logout/index.html -->
+
+<h1>You're logged out</h1>
+<div><a href="/">Log back in</a></div>
+```
+
+When the user refreshes the page, they'll be identified as an unauthorized user, and be redirected to Auth0's login page. For a more advanced implementation of logout functionality, you may choose to _always_ return a redirect to your app's root path: this will force every user to sign in again immediately after logging in:
```

Comment:
```suggestion
When the user refreshes the page, they'll be identified as an unauthorized user, and be redirected to Auth0's login page. For a more advanced implementation of logout functionality, you may choose to _always_ return a redirect to your app's root path: this will force every user to sign in again immediately after logging out:
```

Reply:
oh duh, thanks!